local/compose/convergence.go

```diff
@@ -143,17 +182,30 @@ func nextContainerNumber(containers []moby.Container) (int, error) {
 
 }
 
-func getScale(config types.ServiceConfig) int {
+func getScale(config types.ServiceConfig) (int, error) {
+	scale := 1
+	var err error
 	if config.Deploy != nil && config.Deploy.Replicas != nil {
-		return int(*config.Deploy.Replicas)
+		scale = int(*config.Deploy.Replicas)
 	}
 	if config.Scale != 0 {
-		return config.Scale
+		scale = config.Scale
+	}
+	if scale > 1 && config.ContainerName != "" {
+		scale = -1
+		err = fmt.Errorf(doubledContainerNameWarning,
+			config.Name,
+			config.ContainerName)
 	}
-	return 1
+	return scale, err
 }
 
 func (s *composeService) createContainer(ctx context.Context, project *types.Project, service types.ServiceConfig, name string, number int, autoRemove bool) error {
+	if containerExists(ctx, s.apiClient, name) {
```

Comment:
This will add an inspect api call for every container we create... we should be able to do better